sdk/elixir/lib/dagger_module_runtime.ex

```diff
@@ -0,0 +1,167 @@
+defmodule Dagger.ModuleRuntime do
+  @moduledoc """
+  `Dagger.ModuleRuntime` is a runtime for `Dagger` module for Elixir.
+  """
+
+  def __on_definition__(env, :def, name, args, _guards, _body) do
```

Comment:
To reviewer, the `__on_definition__` invoke when Elixir start compile the source. When it found a function declaration, it'll pass the information to this function.